### DISCUSSION TITLE
CSS with Media Queries in Custom Theme

### DISCUSSION THREAD

**Discussion by baxyz** [keycloak/keycloak#21999] [Q&A] [answered]
Hi,

By using `styles=light.css` within the `theme.property` file, it is not possible to load CSS conditionnaly to a media query.

So, is it ok to use `<link href=${url.resourcesPath}/css/light.css" media="(prefers-color-scheme: light)" rel="stylesheet" type="text/css"/>` directly in the `template.ftl` file?

Additional question: is it good practice to mix `styles=` and `<link/>`?

Thanks for your advice.

**Comment by jonkoops** [comment, marked as answer]
It's no issue modifying `template.ftl` to include your custom styling like this. Just make sure to keep it up-to-date with the upstream when you upgrade Keycloak.

That said, your use-case is very valid. We should probably support this out of the box, feel free to log an issue for it.

**Comment by baxyz** [comment]
Done #22042
